### PR TITLE
Fixes issue with on android with TextInput focus inside ScrollView

### LINKED ISSUE #16635: Unexpected keyboard dismiss in ScrollView when soft keyboard appears.
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

Scanning folders for symlinks in /Users/jiaminglu/test/node_modules (20ms)

Environment:
  OS:  macOS Sierra 10.12.6
  Node:  8.2.1
  Yarn:  0.27.5
  npm:  5.3.0
  Watchman:  4.9.0
  Xcode:  Xcode 9.1 Build version 9B55
  Android Studio:  3.0 AI-171.4408382

Packages:
  react: 16.0.0-beta.5 => 16.0.0-beta.5
  react-native: 0.49.5 => 0.49.5

### Steps to Reproduce

1. Use a TextInput inside a ScrollView, with keyboardDismissMode set to 'on-drag'
2. Layout the TextInput at the bottom of the screen.
3. Click the text input

### Expected Behavior

The keyboard should appear.

### Actual Behavior

The keyboard appears and then disappears immediately.

### Reproducible Demo

1. react-native init test
2. replace App.js with:
```jsx
import React, { Component } from 'react';
import { View, TextInput, ScrollView, Dimensions } from 'react-native';

export default class App extends Component {
  render() {
    return (
      <ScrollView style={{flex: 1}} keyboardDismissMode='on-drag'>
        <View  style={{flex: 1}}>
          <TextInput defaultValue="Hello world" style={{
            marginTop: Dimensions.get('window').height - 50,
            height: 40,
            borderColor: 'black',
            borderWidth: 1,
          }}/>
        </View>
      </ScrollView>
    );
  }
}
```

### Cause

When the TextInput clicked and soft keyboard appeared, the Android system pushes up the underlying EditText, and then the ScrollView detects a scroll event, dismissed the keyboard according to `keyboardDismissMode="on-drag"`, which I think should be handled on `onScrollBeginDrag` instead.

### PR DESCRIPTION
**Pull request by mistenkt**
# Motivation
Fixes #16635
the issue is a little serious if you use a scrollview which contains textinput, you will never get a chance to input something since keyboard will always be dismissed because of on-drag event if your scroll-view use the property: keyboardDismissMode ='on-drag'

# Test Plan
verify the issue #16635 and tested this fix worked

# Release Notes
[ANDROID] [BUGFIX] [ScrollView] - Check that isTouching is also set when dismissing keyboard with on-drag android.

### CODE REVIEW

**Comment by canastro** [comment]
Having this fix merged would be really important for the app I'm currently developing. Any update?

**Comment by canastro** [comment]
Any update on the progress of this merge request?

**Comment by mistenkt** [comment]
Tbh i dont know why this merge supposedly failed on test, i added like 15 characters only. it works fine, and id really like to see this merged as it causes big issued on android without it

**Comment by angly-cat** [comment]
@mistenkt 

> If you don't see anyone comment in a few days feel free to comment mentioning one of the core contributors to the project so they get a notification.

**Comment by mistenkt** [comment]
@sahrens

**Comment by mistenkt** [comment]
Anyone from FB want to tell me why import failed? @hramos

**Comment by hramos** [comment]
I don't see any particular failure that may be keeping this from landing, I'll ping @shergin

**Comment by mistenkt** [comment]
Great!

**Comment by mistenkt** [comment]
Any news on this @hramos ?

**Comment by mistenkt** [comment]
Bumping this again. @hramos @shergin @sahrens

**Comment by hramos** [comment]
I don't review most PRs - there's other contributors in the repo that can do that. Do let me know if a PR needs internal support to land (ie. import failed).

**Comment by hramos** [comment]
Import failed due to a conflict. The PR needs to be rebased.

**Comment by mistenkt** [comment]
Im giving up on this. I literally added 25 characters in one file... And it fixes a breaking bug, but the whole process of getting this merged is crazy..

**Comment by hramos** [comment]
I apologize for the friction. Due to some changes to the internal repo that serves as the source of truth, any PRs opened prior to ~Early February 2018 need to be rebased.

It looks like you had some trouble doing this in this PR. It might be easier to pull from master and create a fresh fork, then apply your small change.

**Comment by hramos** [comment]
Landed in https://github.com/facebook/react-native/pull/18785

File: Libraries/Components/TextInput/InputAccessoryView.js [hunk 1]
```diff
@@ -90,6 +90,8 @@ type Props = {
 
 class InputAccessoryView extends React.Component<Props> {
   render(): React.Node {
+    console.warn('<InputAccessoryView> is not supported on Android yet.');
```
Comment (Ashoat) [inline Libraries/Components/TextInput/InputAccessoryView.js]: @PeteTheHeat, shouldn't this warning be constrained to only occur on Android?

**Comment by axlider** [comment]
I have the same issue with FlatList no matter what I specify for 'keyboardDismissMode'.
Is this related?
My workaround is to scroll my inputfields above the keyboard:
`this.flatList.scrollToIndex({ animated: true, index: index, viewPosition: 0.5});`

**Pull request closed without merging**